sqlcapture/tests/datatypes.go

```diff
@@ -19,6 +19,9 @@ type DatatypeTestCase struct {
 	ExpectType  string
 	InputValue  interface{}
 	ExpectValue string
+
+	PrimaryKeyValue    interface{}
```

Comment:
If it's an either/or choice between primary-key and just-a-value cases, why do we need separate `PrimaryKeyValue` and `PrimaryExpectValue` fields? Couldn't there just be a `PrimaryKey bool` which toggles the behavior while using the same test case fields for input and expected output?

Reply:
It's not an either-or, it's both. When you provide `PrimaryKeyValue`, we test the value both as a normal column and as a primary key. I wasn't sure if there is something to be tested about normal columns that wouldn't apply to primary keys, so I went with the cautious path of doing both.

If doing a primary key check alone is sufficient, I can refactor this to just be a boolean flag.

Reply:
Perhaps I just need more coffee and I'm missing something obvious, but from my reading of the code it really doesn't look like it tests both options. Just as a single example, only one table is created which either has the column of interest as a key or a value. If it were running both options I would expect to see either that or an extra `t.Run()` call somewhere to run separate "as primary key" and "as ordinary value" subtests.

Regardless, I don't think that we need to test both at the same time in the first place. In general there is no difference between how we handle values as primary keys and non primary-keys, and if there were such a difference we could still test it with just a boolean flag by having a pair of test cases like:

```
{ColumnType: "int", ExpectType: `{"type":"integer"}`, InputValue: 123, ExpectValue: `123`},
{ColumnType: "int", ExpectType: `{"type":"integer"}`, InputValue: 123, ExpectValue: `123`, PrimaryKey: true},
```

Reply:
You are right actually, now I remember why I did it this way: we are inserting to the same table from the backfill and the replication parts of the test, but we couldn't insert the same value for a primary key, so I introduced a second value to be unique... However I just refactored this to use a compound key when testing as a primary key, that way we it works with the same value for both backfill and replication and simplifies things a lot

Reply:
Ah, that makes sense. With that change this all looks good to me.